include/datenhaltung.h

```diff
@@ -0,0 +1,20 @@
+#ifndef DATEN_HALTUNG_HEADER
+#define DATEN_HALTUNG_HEADER
+// public
+/*
+    In Result.value steht das value des keys, wenn er gefunden wurde
+    ansonsten bleibt value leer und in Result.error_code steht KEY_NOT_FOUND
+*/
+Result find_by_key(char* key);
+
+/*
+    Gibt SUCCESS oder eine Fehlermeldung über 1000 zurück
+*/
+int save(char* key, char* value);
+
+/*
+    Gibt SUCCESS oder eine Fehlermeldung über 1000 zurück
+*/
+int delete_by_key(char* key);
+
+#endif
```

Comment:
When I see this correctly, `datenhaltung.h` is just copy&pasted from the main branch. This causes a merge conflict and is generally not a good idea.
It is better to merge or cherry-pick the desired changes.
So you need to revert commit 75d7f42e and merge the main branch into this one.